students/bcoates/lesson03/mailroom.py

```diff
@@ -0,0 +1,136 @@
+#!/usr/bin/env python3
+
+# Starting list of donors and their donation amounts
+donor_history = [
+    ["Lana Kane", [2999.99]],
+    ["Cheryl Tunt", [150.20, 98192.10]],
+    ["Cyril Figgis", [819.25, 998.62, 10.50]],
+    ["Pam Poovey", [74926.10, 2675.87, 3289.33]],
+    ["Ray Gillette", [3820.90]]
+]
+
+def main_menu():
+    """ Prompt a user to selection an action to take, 
+    and call the appropriate function """
+    
+    # Prompt the user and set the selection
+    while True:
+        selection = input("""
+Please enter a number to select the action:
+1 - Send a Thank You
+2 - Create a Report
+3 - quit
+> """)
+
+        # Evaluate the selection
+        if int(selection) == 1:
+            print("Let's send a thank you!")
+            send_a_thank_you()
+        elif int(selection) == 2:
+            print("Let's create a report!")
+            create_a_report()
+        elif int(selection) == 3:
+            print("Quitting...")
+            break
+        else:
+            print("Please select a valid number...")
+
+def prompt_donor_name():
+    """ Prompt for a donor name, list if needed, add if not found """
+
+    while True:
+        # Prompt the user for the full name
+        full_name = input("Please enter the full name of a donor (type 'list' for current donor names) > ")
+
+        # If the name is 'list' print the full list of names
+        if full_name.lower() == "list":
+            for donors in donor_history:
+                print(donors[0])
+        elif full_name == "":
+            return "NODATA"
+        else:
+            # Check for existing name in the list
+            found_match = False
+            for donors in donor_history:
+                if full_name in donors:
+                    print("Found existing donor...")
+                    found_match = True
+                    return full_name
+            # If not match found, add new
+            if found_match == False:
```

Comment:
you can actually use `else` statement with `for` loop to avoid using `found_match`

here is suggestion for more pythonic code:
```
for donor in donor_history:
    if full_name == donor[0]:
        print("Found existing donor")
        return donor
else:
    print("Adding new donor")
    new_donor = [full_name, []]
    donor_history.append(new_donor)
    return new_donor
```